projects/user-telemetry-client/src/TelemetryService.ts

```diff
@@ -2,12 +2,31 @@ import throttle from 'lodash/throttle';
 import chunk from 'lodash/chunk';
 
 import { IUserTelemetryEvent } from "./";
+import { createHmac } from "crypto";
+
+export function generateHmacHeaders(requestDate: Date, path: string, hmacSecretKey: string): {
+    ["x-gu-tools-hmac-token"]: string,
+    ["x-gu-tools-hmac-date"]: string
+} {
+    const date = requestDate.toUTCString()
+
+    const hmac = createHmac("sha256", hmacSecretKey);
+    const content = date + "\n" + path;
+    hmac.update(content, "utf8");
+
+    const token = "HMAC " + hmac.digest("base64");
+
+    return {
+        ["x-gu-tools-hmac-token"]: token,
+        ["x-gu-tools-hmac-date"]: date
+    }
+}
 
 export class UserTelemetryEventSender {
     private postEventLimit = 500;
     private eventBuffer: IUserTelemetryEvent[] = [];
 
-    public constructor(private telemetryUrl: string, private throttleDelay: number = 10000) {};
+    public constructor(private telemetryUrl: string, private throttleDelay: number = 10000, private useHmacSecret: false | string = false) {};
```

Comment:
might be worth adding an annotation to the class with some guidance on when the hmacSecret  should be used and how to manage it.

Reply:
Great point, I have added a comment: https://github.com/guardian/editorial-tools-user-telemetry-service/pull/50/commits/22931124782bad0430c9c9835d60d8613f960fcf